src/update-feeds/update-feeds.ts

```diff
@@ -128,42 +130,66 @@ export const runUpdateFeed = async (providerName: Provider, chain: Chain, chainI
   });
 };
 
-export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponse[]) =>
-  batch.map((dapiResponse: ReadDapiWithIndexResponse) => {
-    const signedData = getStoreDataPoint(dapiResponse.decodedDataFeed.dataFeedId);
-    if (!signedData) {
-      return { ...dapiResponse, shouldUpdate: false };
-    }
+// https://github.com/api3dao/airnode-protocol-v1/blob/fa95f043ce4b50e843e407b96f7ae3edcf899c32/contracts/api3-server-v1/DataFeedServer.sol#L132
+export const decodeBeaconValue = (encodedBeaconValue: string) => {
+  const decodedBeaconValue = ethers.BigNumber.from(
+    ethers.utils.defaultAbiCoder.decode(['int256'], encodedBeaconValue)[0]
+  );
+  if (decodedBeaconValue.gt(INT224_MAX) || decodedBeaconValue.lt(INT224_MIN)) {
+    return null;
+  }
 
-    const offChainValue = ethers.BigNumber.from(signedData.encodedValue);
-    const offChainTimestamp = Number.parseInt(signedData?.timestamp ?? '0', 10);
-    const deviationThreshold = dapiResponse.updateParameters.deviationThresholdInPercentage;
-    const shouldUpdate = checkUpdateConditions(
-      dapiResponse.dataFeedValue.value,
-      dapiResponse.dataFeedValue.timestamp,
-      offChainValue,
-      offChainTimestamp,
-      dapiResponse.updateParameters.heartbeatInterval,
-      deviationThreshold
-    );
+  return decodedBeaconValue;
+};
 
-    return {
-      ...dapiResponse,
-      signedData,
-      shouldUpdate,
-    };
-  });
+export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponse[]): UpdateableDapi[] =>
```

Comment:
@vponline I removed the `shouldUpdate` flag because I want this function to return the signed data for each beacon (so that we can use that data to update them) but if there is no data for a beacon there is nothing to return. I disliked making a new type that would have signed data optional so I just filtered the invalid values here directly.